rust/operator-binary/src/util.rs

```diff
@@ -0,0 +1,59 @@
+use snafu::{OptionExt, Snafu};
+use stackable_airflow_crd::AirflowCluster;
+use stackable_operator::k8s_openapi::api::batch::v1::Job;
+use stackable_operator::k8s_openapi::api::core::v1::{EnvVar, EnvVarSource, SecretKeySelector};
+
+#[derive(Snafu, Debug)]
+#[allow(clippy::enum_variant_names)]
+#[snafu(context(suffix(false)))]
+pub enum Error {
+    #[snafu(display("object defines no version"))]
+    ObjectHasNoVersion,
+}
+
+pub enum JobState {
+    InProgress,
+    Complete,
+    Failed,
+}
+
+pub fn get_job_state(job: &Job) -> JobState {
+    let conditions = job
+        .status
+        .as_ref()
+        .and_then(|status| status.conditions.clone())
+        .unwrap_or_default();
+
+    if conditions
+        .iter()
+        .any(|condition| condition.type_ == "Failed" && condition.status == "True")
+    {
+        JobState::Failed
+    } else if conditions
+        .iter()
+        .any(|condition| condition.type_ == "Complete" && condition.status == "True")
+    {
+        JobState::Complete
+    } else {
+        JobState::InProgress
+    }
+}
+
+pub fn airflow_version(airflow: &AirflowCluster) -> Result<&str, Error> {
```

Comment:
I would have expected this to be a method of `impl AirflowCluster`.

Reply:
Done - moved to AirflowCluster